src/files.ml

```diff
@@ -3,6 +3,211 @@
 open Extra
 open Console
 
+(** Logging function for evaluation. *)
+let log_file = new_logger 'f' "file" "file system"
+let log_file = log_file.logger
+
+(** Representation of module paths and related operations. *)
+module Path =
+  struct
+    (** Representation of a module path (roughly, a file path). *)
+    type module_path = string list
+
+    (** Short synonym of [module_path]. *)
+    type t = module_path
+
+    (** [compare] is a standard comparing function for module paths. *)
+    let compare : t -> t -> int = Pervasives.compare
+
+    (** [pp oc mp] prints [mp] to channel [oc]. *)
+    let pp : module_path pp = fun oc mp ->
+      Format.pp_print_string oc (String.concat "." mp)
+  end
+
+(** Functional maps with module paths as keys. *)
+module PathMap = Map.Make(Path)
+
+(** Synonym of [string] for file paths. *)
```

Comment:
Wouldn't be better before [Path] module definition?

Reply:
Some times you use string, and sometimes file_path. Why not using string (or file_path) everywhere?

Reply:
> Wouldn't be better before [Path] module definition?

No, I dont think so.

> Some times you use string, and sometimes file_path. Why not using string (or file_path) everywhere?

I'm only using `file_path` where an actual file path is expected. A module path member is definitely not a file path, so it is normal to not use `file_path` everywhere.

Reply:
Sure but, later, you do not use file_path anymore. For instance, `lib_root : string option Pervasives.ref`.